### DISCUSSION TITLE
Are "Ignored" Class Labels for 2D Detection Ignored or Mapped to Other Classes?

### DISCUSSION THREAD

**Discussion by CJMenart** [bdd100k/bdd100k#291] [Q&A] [answered]
The question is prompted mainly by the evaluation config file at: https://github.com/bdd100k/bdd100k/blob/master/bdd100k/configs/det.toml

Three classes, "other vehicle", "other person", and "trailer" are marked as ignored classes for 2D Detection. That's consistent with the description of this task in the docs (https://doc.bdd100k.com/format.html).

Where I get thrown off a bit is the presence of "mappings" for these three classes. If these classes were ignored, why would they have a mapping? Are labels for these classes discarded during detection evaluation, or are they mapped to instances of the 10 classes that are used, i.e. evaluating an "other vehicle" instance as if it is a car?

I've tried to dig through the code a bit but the evaluation routes through Pycocotools and whatnot, so I haven't been able to figure it out yet. Maybe someone here knows. My guess is that these labels get tossed, but for training purposes it would be nice to know for sure.

**Comment by tobiasfshr** [comment, marked as answer]
Hi, thanks for your interest in the dataset. The important piece of code is here: https://github.com/scalabel/scalabel/blob/9784a1dff92e17c63ea13f6f33b1a024f032d4f9/scalabel/label/to_coco.py#L183
As you can see, even if a box is to be ignored, it still needs a category id, so we have the mapping in order to tell the COCO eval which class this box is an ignore region for. E.g. detecting a pedestrian in an area labeled as trailer would not be correct, but truck would be okay.
